cmd/server-main.go

```diff
@@ -310,6 +310,30 @@ func serverMain(ctx *cli.Context) {
 	globalObjectAPI = newObject
 	globalObjLayerMutex.Unlock()
 
+	err = globalObjectAPI.FixDeploymentID(context.Background())
```

Comment:
Can we avoid adding "FixDeploymentID" to object layer interface and call deployment fix code in newObjectLayer(), ?